cmake: Replace deprecated `qt6_add_translation` with `qt6_add_lrelease`

hebasto opened the pull request:
See Qt docs:
- https://doc.qt.io/archives/qt-6.2/qtlinguist-cmake-qt-add-translation.html
- https://doc.qt.io/archives/qt-6.2/qtlinguist-cmake-qt-add-lrelease.html

fanquake commented on src/qt/CMakeLists.txt:
```diff
@@ -250,6 +243,16 @@ if(qt_lib_type STREQUAL "STATIC_LIBRARY")
   )
 endif()
 
+# TODO: The file(GLOB ...) command should be replaced with an explicit
```
If we're touching it already, can we fix the TODO, rather than move it? Or, put it into an issue, so that someone might pick it up?

hebasto replied:
> Or, put it into an issue, so that someone might pick it up?

https://github.com/bitcoin/bitcoin/issues/32653.

fanquake replied:
Delete it from here now?

hebasto replied:
Sure! The comment has been deleted.

hebasto commented:
My Guix build:
```
aarch64
58c6cdfd914bfd2f2646640166692a0a02ec515809d5fa05cb3c9ff511617958  guix-build-d73400e06bef/output/aarch64-linux-gnu/SHA256SUMS.part
51fdf3f3ddb16b5aaf571f33f0da5b95f0b728c7d07863df48dfd3c2d02480dc  guix-build-d73400e06bef/output/aarch64-linux-gnu/bitcoin-d73400e06bef-aarch64-linux-gnu-debug.tar.gz
cd2198dbde7ea6af923cc24c4a7363a3355e18f14a397994684db3c1b1662108  guix-build-d73400e06bef/output/aarch64-linux-gnu/bitcoin-d73400e06bef-aarch64-linux-gnu.tar.gz
855c93e3337f94d94c812a4c17d1345f92b59ecc1f8f1aac0b7ea91c41ba7a28  guix-build-d73400e06bef/output/arm-linux-gnueabihf/SHA256SUMS.part
381c366308687b798a832e49977b3c6091d47083a5537f324a5289ed56a74be0  guix-build-d73400e06bef/output/arm-linux-gnueabihf/bitcoin-d73400e06bef-arm-linux-gnueabihf-debug.tar.gz
db42113bc4b008a8206b4f059e910c4c354df7052c59f5bc666288383ddedb39  guix-build-d73400e06bef/output/arm-linux-gnueabihf/bitcoin-d73400e06bef-arm-linux-gnueabihf.tar.gz
6d0b9920983bcee90aeaa05e8ab0d6f4c4bc11d97f9c69d719165d613e412962  guix-build-d73400e06bef/output/arm64-apple-darwin/SHA256SUMS.part
5e4219bd55927b11e27de324236f96a8d450ff8c9c689a860a9d4f2d5f588c64  guix-build-d73400e06bef/output/arm64-apple-darwin/bitcoin-d73400e06bef-arm64-apple-darwin-codesigning.tar.gz
76a2ced5cfbb6887f9977137f16503098d84e527b93d5851c423a5acc7fcc44c  guix-build-d73400e06bef/output/arm64-apple-darwin/bitcoin-d73400e06bef-arm64-apple-darwin-unsigned.tar.gz
ef5ecb1438895c026a0a16a340851e811b71f821206011c9fc5d48beba293914  guix-build-d73400e06bef/output/arm64-apple-darwin/bitcoin-d73400e06bef-arm64-apple-darwin-unsigned.zip
426230a49fa013a3f8048892f0d23ec25361b97f909eda67bf1b0d7e26cdb08e  guix-build-d73400e06bef/output/dist-archive/bitcoin-d73400e06bef.tar.gz
a7c7a10a6e716f808d476595990fa27a3df7a035214a59d9e8653cfc4ba47c0e  guix-build-d73400e06bef/output/powerpc64-linux-gnu/SHA256SUMS.part
97b36a9282c0140af8b635aa34ebebc2d4a70fe3092d9dde4c760bc8093b0eb6  guix-build-d73400e06bef/output/powerpc64-linux-gnu/bitcoin-d73400e06bef-powerpc64-linux-gnu-debug.tar.gz
ef5af0e9f941919ef53f5ca06ec54df410c2cb555080f489a8c2309ddbcfbdb3  guix-build-d73400e06bef/output/powerpc64-linux-gnu/bitcoin-d73400e06bef-powerpc64-linux-gnu.tar.gz
faa17b0f004b72758f9075fb7ceafcad73e6b2e2ed91c53cdd807a83ec995689  guix-build-d73400e06bef/output/riscv64-linux-gnu/SHA256SUMS.part
e5320db6379de72be50c516b1a6bf2f74f4d92078ff443aafd2348ca80687c9f  guix-build-d73400e06bef/output/riscv64-linux-gnu/bitcoin-d73400e06bef-riscv64-linux-gnu-debug.tar.gz
9f05585f9807fcfde2626f3927ea17e236fcf5fa8096d1deaac76667d92b759f  guix-build-d73400e06bef/output/riscv64-linux-gnu/bitcoin-d73400e06bef-riscv64-linux-gnu.tar.gz
f867a6e4095cfd369abf99fd46167e7b07f3fbf7d1637965109ffd04af1fbf65  guix-build-d73400e06bef/output/x86_64-apple-darwin/SHA256SUMS.part
a55f118bf5bedcc397bbe39096fecfa56a5ec8af93828dd8f337e3c516439a57  guix-build-d73400e06bef/output/x86_64-apple-darwin/bitcoin-d73400e06bef-x86_64-apple-darwin-codesigning.tar.gz
906d090d7c064e07fe524bededbd39aee957addad45f4391b0a1d5a7b351becb  guix-build-d73400e06bef/output/x86_64-apple-darwin/bitcoin-d73400e06bef-x86_64-apple-darwin-unsigned.tar.gz
ae3c4eca3d0f82f6c05e58d11be0e69c967661b47d4f2ab113a6b6da35ab894f  guix-build-d73400e06bef/output/x86_64-apple-darwin/bitcoin-d73400e06bef-x86_64-apple-darwin-unsigned.zip
5213c02e1e6b0749253f124512a1204c7e13c88e1579b81fd4d74d482cc1f278  guix-build-d73400e06bef/output/x86_64-linux-gnu/SHA256SUMS.part
4d763ddda9359e8efbcf54761e50f5aaf3440cdd05a1709e1efe1ae54b07c3d2  guix-build-d73400e06bef/output/x86_64-linux-gnu/bitcoin-d73400e06bef-x86_64-linux-gnu-debug.tar.gz
d01a16daf2ecb04b6d0f4fe447564ba5f68c037865290cee2be9e42d8c8e260c  guix-build-d73400e06bef/output/x86_64-linux-gnu/bitcoin-d73400e06bef-x86_64-linux-gnu.tar.gz
ff7ad99e2ff2b6e25e9f45610066229f0048f746f31d397af12229ff9b55cb06  guix-build-d73400e06bef/output/x86_64-w64-mingw32/SHA256SUMS.part
dca72d8dab219afd51d610267089cf6dfad66c890bb6e7b502c97ae2f47193a8  guix-build-d73400e06bef/output/x86_64-w64-mingw32/bitcoin-d73400e06bef-win64-codesigning.tar.gz
62390a20f649c9fa546f6508630ec504468569c31e2e7b3df568ffb2292b8f96  guix-build-d73400e06bef/output/x86_64-w64-mingw32/bitcoin-d73400e06bef-win64-debug.zip
6b43e6e793ce2f2cf9b7096f555ce227606c07fc15300ed71ec43a6ba9dcf81e  guix-build-d73400e06bef/output/x86_64-w64-mingw32/bitcoin-d73400e06bef-win64-setup-unsigned.exe
1bbe4539e3935d4cdc2ec085d3a4e617e2b5b6cbfac0407d7119cb3258b91180  guix-build-d73400e06bef/output/x86_64-w64-mingw32/bitcoin-d73400e06bef-win64-unsigned.zip
```

hebasto committed:
cmake: Replace deprecated `qt6_add_translation` with `qt6_add_lrelease`

See Qt docs:
- https://doc.qt.io/archives/qt-6.2/qtlinguist-cmake-qt-add-translation.html
- https://doc.qt.io/archives/qt-6.2/qtlinguist-cmake-qt-add-lrelease.html

hebasto commented:
> From the documentation i understand that it's not `qt_add_translations` itself that was deprecated, but the specific syntax of using it. Starting with Qt 6.7, it uses a different format, but it won't be removed? It's slightly ambigious to me, tbh.

`qt_add_translation` (singular) is deprecated.  `qt_add_translations` (plural) is a new Qt 6 command that combines `lupdate` and `lrelease` invocations.

laanwj commented:
> qt_add_translation (singular) is deprecated. qt_add_translations (plural) is a new Qt 6 command that combines the lupdate and lrelease invocations.

Right, and we only need `lrealease` here. Okay, concept ACK.

fanquake reviewed (approved):
ACK 18cf727429e967053d6a7468c9a40dc14185af34
